lsp-types/src/Language/LSP/Protocol/Types/CustomInstances.hs

```diff
@@ -0,0 +1,27 @@
+{-# OPTIONS_GHC -Wno-orphans #-}
+module Language.LSP.Protocol.Types.CustomInstances where
+
+import           Language.LSP.Protocol.Internal.Types
+import           Data.Default
+import           Data.Hashable
+import           Data.Semigroup                ()
+
+instance Semigroup WorkspaceEdit where
```

Comment:
suggest putting these in `Edit.hs`

Reply:
If we want to export instances together with Types, than Capabilties will need to stay here (or move to Types/Capabilities.hs?!). If they do stay here, it wouldn't be immediately obvious at first where new instances should be placed, with some being with their category and some being here.

Reply:
Hmm, okay, that does make sense. In that case I suggest renaming this to "Orphans.hs" (which is a common convention for modules of extra instances that aren't with their types), and maybe we could move some of the other orphan instances in nearby files into it too.